runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3060,14 +3060,39 @@ hookAcquireVMAccess(J9HookInterface** hook, UDATA eventNum, void* voidEventData,
 {
 	J9VMAcquireVMAccessEvent* eventData = (J9VMAcquireVMAccessEvent*)voidEventData;
 
-	J9VMThread *vmThread = eventData->currentThread;
-	MM_EnvironmentStandard *env = MM_EnvironmentStandard::getEnvironment(vmThread->omrVMThread);
-	MM_GCExtensions* ext = MM_GCExtensions::getExtensions(vmThread);
+	J9VMThread *currentThread = eventData->currentThread;
+	MM_EnvironmentStandard *env = MM_EnvironmentStandard::getEnvironment(currentThread->omrVMThread);
+	MM_GCExtensions* ext = MM_GCExtensions::getExtensions(currentThread);
 
 	Assert_MM_true(ext->concurrentScavenger);
 
 	ext->scavenger->switchConcurrentForThread(env);
 }
+
+static void
+hookReleaseVMAccess(J9HookInterface** hook, UDATA eventNum, void* voidEventData, void* userData)
+{
+	J9VMReleaseVMAccessEvent* eventData = (J9VMReleaseVMAccessEvent*)voidEventData;
+
+	J9VMThread *currentThread = eventData->currentThread;
+	MM_GCExtensions* ext = MM_GCExtensions::getExtensions(currentThread);
+
+	if (ext->isConcurrentScavengerInProgress()) {
+	}
```

Comment:
Something missing here?  If its waiting on OMR functionality there should be a comment as a reminder